### ISSUE TITLE
No file or line history; file path is appended to repo path as absolute, not relative, path (regression in 10.2.2)

### ISSUE THREAD

**Issue by fsackur** [eamodio/vscode-gitlens#1090]
- GitLens Version: 10.2.2
- VSCode Version: 1.47.3
- OS Version: Win 10 (2004)

Issue does not occur in 10.2.1.

Steps to Reproduce:

1. Open code normally
2. Open tracked file in editor
3. File and line history show `No file history could be found.` / `No line history could be found.`

Output shows that the absolute path is being appended to the repo path: 
```
[2020-08-07 11:20:51:050] [1a5] GitService.getLogForFile Skipping log; 'c:\githubdata\windows-update\c:\Githubdata\windows-update\Public\Install-WindowsUpdates.ps1' is not tracked
```

Full output:
```
[2020-08-07 11:20:51:035] [c:\Githubdata\windows-update\Public] git rev-parse --show-toplevel • 69 ms 
[2020-08-07 11:20:51:035] [c:\Githubdata\windows-update\Public] git rev-parse --show-toplevel • 56 ms (await) 
[2020-08-07 11:20:51:035] [18d] GitService.getRepoPathCore completed • 69 ms
[2020-08-07 11:20:51:035] [195] GitService.getRepoPathCore completed • 57 ms
[2020-08-07 11:20:51:035] [18b] GitService.getRepoPath returned c:/githubdata/windows-update • 70 ms
[2020-08-07 11:20:51:035] [193] GitService.getRepoPath returned c:/githubdata/windows-update • 57 ms
[2020-08-07 11:20:51:035] [18a] GitUri.fromUri returned GitUri(file:///c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1 repoPath=c:/githubdata/windows-update) • 70 ms
[2020-08-07 11:20:51:035] [196] FileHistoryTrackerNode.resetChild
[2020-08-07 11:20:51:035] [196] FileHistoryTrackerNode.resetChild completed • 0 ms
[2020-08-07 11:20:51:035] [192] GitUri.fromUri returned GitUri(file:///c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1 repoPath=c:/githubdata/windows-update) • 57 ms
[2020-08-07 11:20:51:035] [197] LineHistoryTrackerNode.resetChild
[2020-08-07 11:20:51:035] [197] LineHistoryTrackerNode.resetChild completed • 0 ms
[2020-08-07 11:20:51:035] [189] FileHistoryTrackerNode.refresh returned false, uri=GitUri(file:///c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1 repoPath=c:/githubdata/windows-update) • 71 ms
[2020-08-07 11:20:51:035] [191] LineHistoryTrackerNode.refresh returned false, uri=GitUri(file:///c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1 repoPath=c:/githubdata/windows-update) • 57 ms
[2020-08-07 11:20:51:035] [198] FileHistoryView.triggerNodeChange — e=FileHistoryTrackerNode
[2020-08-07 11:20:51:035] [198] FileHistoryView.triggerNodeChange completed • 0 ms
[2020-08-07 11:20:51:036] [199] LineHistoryView.triggerNodeChange — e=LineHistoryTrackerNode
[2020-08-07 11:20:51:036] [199] LineHistoryView.triggerNodeChange completed • 0 ms
[2020-08-07 11:20:51:036] [188] FileHistoryView.refreshNode completed • 71 ms
[2020-08-07 11:20:51:036] [187] FileHistoryTrackerNode.triggerChange completed • 71 ms
[2020-08-07 11:20:51:036] [190] LineHistoryView.refreshNode completed • 58 ms
[2020-08-07 11:20:51:036] [18f] LineHistoryTrackerNode.triggerChange completed • 58 ms
[2020-08-07 11:20:51:038] [19a] FileHistoryView.getNodeLastKnownLimit completed • 0 ms — e=FileHistoryNode(gitlens:repository(c:/githubdata/windows-update):history:file(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1))
[2020-08-07 11:20:51:038] [19b] LineHistoryView.getNodeLastKnownLimit completed • 0 ms — e=LineHistoryNode(gitlens:repository(c:/githubdata/windows-update):history:line(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1[188,13-188,13]))
[2020-08-07 11:20:51:038] [19c] FileHistoryNode(gitlens:repository(c:/githubdata/windows-update):history:file(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1)).ensureSubscription
[2020-08-07 11:20:51:038] [19d] FileHistoryNode(gitlens:repository(c:/githubdata/windows-update):history:file(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1)).subscribe
[2020-08-07 11:20:51:038] [19e] GitService.getRepository — e=GitUri(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1 repoPath=c:/githubdata/windows-update)
[2020-08-07 11:20:51:038] [19f] LineHistoryNode(gitlens:repository(c:/githubdata/windows-update):history:line(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1[188,13-188,13])).ensureSubscription
[2020-08-07 11:20:51:038] [1a0] LineHistoryNode(gitlens:repository(c:/githubdata/windows-update):history:line(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1[188,13-188,13])).subscribe
[2020-08-07 11:20:51:038] [1a1] GitService.getRepository — e=GitUri(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1 repoPath=c:/githubdata/windows-update)
[2020-08-07 11:20:51:039] [19e] GitService.getRepository returned c:/githubdata/windows-update • 0 ms
[2020-08-07 11:20:51:039] [1a1] GitService.getRepository returned c:/githubdata/windows-update • 0 ms
[2020-08-07 11:20:51:039] [19d] FileHistoryNode(gitlens:repository(c:/githubdata/windows-update):history:file(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1)).subscribe completed • 0 ms
[2020-08-07 11:20:51:039] [1a0] LineHistoryNode(gitlens:repository(c:/githubdata/windows-update):history:line(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1[188,13-188,13])).subscribe completed • 0 ms
[2020-08-07 11:20:51:039] [19c] FileHistoryNode(gitlens:repository(c:/githubdata/windows-update):history:file(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1)).ensureSubscription completed • 0 ms
[2020-08-07 11:20:51:039] [19f] LineHistoryNode(gitlens:repository(c:/githubdata/windows-update):history:line(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1[188,13-188,13])).ensureSubscription completed • 0 ms
[2020-08-07 11:20:51:041] [1a2] GitService.getStatusForFile — e=c:/githubdata/windows-update, t=c:\githubdata\windows-update\c:\Githubdata\windows-update\Public\Install-WindowsUpdates.ps1
[2020-08-07 11:20:51:049] [1a3] GitService.getBlameForRange — e=GitUri(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1 repoPath=c:/githubdata/windows-update), t={"start":{"line":188,"character":13},"end":{"line":188,"character":13},"active":{"line":188,"character":13},"anchor":{"line":188,"character":13}}
[2020-08-07 11:20:51:049] [1a4] GitService.getBlameForFile — e=GitUri(file:///c:/githubdata/windows-update/c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1 repoPath=c:/githubdata/windows-update)
[2020-08-07 11:20:51:049] [1a4] GitService.getBlameForFile Cache hit: 'blame'
[2020-08-07 11:20:51:049] [1a4] GitService.getBlameForFile completed • 0 ms
[2020-08-07 11:20:51:049] [1a3] GitService.getBlameForRange completed • 0 ms
[2020-08-07 11:20:51:049] [1a5] GitService.getLogForFile — e=c:/githubdata/windows-update, t=c:\githubdata\windows-update\c:\Githubdata\windows-update\Public\Install-WindowsUpdates.ps1, i={"limit":10,"range":{"start":{"line":188,"character":13},"end":{"line":188,"character":13},"active":{"line":188,"character":13},"anchor":{"line":188,"character":13}}}
[2020-08-07 11:20:51:050] [1a6] GitService.isTracked returned false • 0 ms — e=c:\githubdata\windows-update\c:\Githubdata\windows-update\Public\Install-WindowsUpdates.ps1, t=c:/githubdata/windows-update, i={}
[2020-08-07 11:20:51:050] [1a5] GitService.getLogForFile Skipping log; 'c:\githubdata\windows-update\c:\Githubdata\windows-update\Public\Install-WindowsUpdates.ps1' is not tracked
[2020-08-07 11:20:51:050] [1a5] GitService.getLogForFile completed • 0 ms
[2020-08-07 11:20:51:107] [c:/githubdata/windows-update] git status --porcelain=v2 --find-renames -- c:/Githubdata/windows-update/Public/Install-WindowsUpdates.ps1 • 65 ms 
[2020-08-07 11:20:51:107] [1a7] GitStatusParser.parse completed • 0 ms
[2020-08-07 11:20:51:107] [1a2] GitService.getStatusForFile completed • 65 ms
[2020-08-07 11:20:51:107] [1a8] GitService.getLogForFile — e=c:/githubdata/windows-update, t=c:\githubdata\windows-update\c:\Githubdata\windows-update\Public\Install-WindowsUpdates.ps1, i={"limit":10}
[2020-08-07 11:20:51:108] [1a8] GitService.getLogForFile Cache hit: 'log:n10:follow'
[2020-08-07 11:20:51:108] [1a8] GitService.getLogForFile completed • 0 ms
```

**Comment by fsackur** [comment]
This is really driving me nuts! Please take that as a reflection on the quality and power of GitLens ;-)

I believe that this would fix:
https://github.com/fsackur/vscode-gitlens/compare/master...fsackur:absolute-path-bug?expand=1

I am in now way set up to take this PR on, so would be very grateful if an experienced maintainer could take this and run with it.

**Comment by fsackur** [comment]
Aha! Related: #932 (I just spotted in the changelog as fixed in 10.2.1)

This duplicates #1045 which contains a link to a beta version.

**Comment by fsackur** [comment]
Closing as duplicate.

I find the file and line history generally stable in the beta - thanks, contributors!
